soccerdata/whoscored.py

```diff
@@ -706,7 +706,7 @@ def read_events(  # noqa: C901
                     var="require.config.params['args'].matchCentreData",
                     no_cache=live,
                 )
-                if retry_missing and reader.read(4) == b"null":
+                if retry_missing and reader.read() == b"null" or reader.read() == b"":
```

Comment:
I've since noticed that this line forces a retry because the latter `reader.read()` always equates to true. This is because the cursor has moved to the end of the reader. I've added a suggestion for you below:

```suggestion
                reader_value = reader.read()
                if retry_missing and reader_value == b"null" or reader_value == b"":
```